tmt/steps/provision/__init__.py

```diff
@@ -317,31 +365,43 @@ class GuestFacts(SerializableContainer):
         default=cast(Optional['tmt.package_managers.GuestPackageManager'], None))
 
     has_selinux: Optional[bool] = None
+    has_systemd: Optional[bool] = None
     is_superuser: Optional[bool] = None
     is_ostree: Optional[bool] = None
 
-    #: Various Linux capabilities and whether they are permitted to
-    #: commands executed on this guest.
-    capabilities: dict[GuestCapability, bool] = field(
-        default_factory=cast(Callable[[], dict[GuestCapability, bool]], dict),
-        serialize=lambda capabilities: {
-            capability.value: enabled
-            for capability, enabled in capabilities.items()
-            } if capabilities else {},
-        unserialize=lambda raw_value: {
-            GuestCapability(raw_capability): enabled
-            for raw_capability, enabled in raw_value.items()
-            }
-        )
+    #: Permitted Linux capabilities on this guest.
+    capabilities: Optional[list[GuestCapability]] = field(
+        default=None,
+        serialize=lambda caps: [cap.value for cap in caps] if caps else None,
+        unserialize=lambda serialized: [GuestCapability(value) for value in serialized]
```

Comment:
Just a nitpicking: IIUIC the code here, it'd emit a list of integers, which wouldn't be nice to read by humans. Maybe we should emit the names instead of values here, it would be a bit more work when unserializing, but inspecting the on-disk files would be easier for puny humans.

Reply:
I've initially had a `Final[list[str]]`, but creating a type, `Literal[*<the final list>]` is not possible on Python <3.10 and while `Literal[tuple(<final list>)] does work, mypy can't deal with it.  Using `eumerate` didn't feel right to me, as it's a bit of black box, but I'd be happy to implement whatever you say is the best way @happz :)